htdocs/adherents/class/adherent.class.php

```diff
@@ -714,109 +714,108 @@ function update_end_date($user)
+		dol_syslog(get_class($this)."::delete", LOG_DEBUG);
+		$resql=$this->db->query($sql);
+		if (! $resql)
+		{
+			$error++;
+			$this->error .= $this->db->lasterror();
+			$errorflag=-1;
+		}
 
-        }
+		// Remove subscription
+		if (! $error)
+		{
+			$sql = "DELETE FROM ".MAIN_DB_PREFIX."subscription WHERE fk_adherent = ".$rowid;
+			dol_syslog(get_class($this)."::delete", LOG_DEBUG);
+			$resql=$this->db->query($sql);
+			if (! $resql)
+			{
+				$error++;
+				$this->error .= $this->db->lasterror();
+				$errorflag=-2;
+			}
+		}
 
-        // Remove subscription
-        if (! $error)
-        {
-        	 $sql = "DELETE FROM ".MAIN_DB_PREFIX."subscription WHERE fk_adherent = ".$rowid;
-        	dol_syslog(get_class($this)."::delete", LOG_DEBUG);
-        	$resql=$this->db->query($sql);
-        	if (! $resql)
-        	{
-        		$error++;
-        		$this->error .= $this->db->lasterror();
-        		$errorflag=-2;
-        	}
-        }
+		// Remove linked user
+		if (! $error)
+		{
+			$ret=$this->setUserId(0);
+			if ($ret < 0)
+			{
+				$error++;
+				$this->error .= $this->db->lasterror();
+				$errorflag=-3;
+			}
+		}
 
-        // Remove linked user
-        if (! $error)
-        {
-        	$ret=$this->setUserId(0);
-        	if ($ret < 0)
-        	{
-        		$error++;
-        		$this->error .= $this->db->lasterror();
-        		$errorflag=-3;
-        	}
-        }
+		// Removed extrafields
+		if (! $error)
+		{
+			if (empty($conf->global->MAIN_EXTRAFIELDS_DISABLED)) // For avoid conflicts if trigger used
+			{
+				$result=$this->deleteExtraFields();
+				if ($result < 0)
+				{
+					$error++;
+					$errorflag=-4;
+					dol_syslog(get_class($this)."::delete erreur ".$errorflag." ".$this->error, LOG_ERR);
+				}
+			}
+		}
 
-        // Removed extrafields
-        if (! $error)
-        {
-        	if (empty($conf->global->MAIN_EXTRAFIELDS_DISABLED)) // For avoid conflicts if trigger used
-        	{
-        		$result=$this->deleteExtraFields();
-        		if ($result < 0)
-        		{
-        			$error++;
-        			$errorflag=-4;
-        			dol_syslog(get_class($this)."::delete erreur ".$errorflag." ".$this->error, LOG_ERR);
-        		}
-        	}
-        }
+		// Remove adherent
+		if (! $error)
+		{
+			$sql = "DELETE FROM ".MAIN_DB_PREFIX."adherent WHERE rowid = ".$rowid;
+			dol_syslog(get_class($this)."::delete", LOG_DEBUG);
+			$resql=$this->db->query($sql);
+			if (! $resql)
+			{
+				$error++;
+				$this->error .= $this->db->lasterror();
+				$errorflag=-5;
+			}
+		}
 
-        // Remove adherent
-        if (! $error)
-        {
-        	$sql = "DELETE FROM ".MAIN_DB_PREFIX."adherent WHERE rowid = ".$rowid;
-        	dol_syslog(get_class($this)."::delete", LOG_DEBUG);
-        	$resql=$this->db->query($sql);
-        	if (! $resql)
-        	{
-        		$error++;
-        		$this->error .= $this->db->lasterror();
-        		$errorflag=-5;
-        	}
-        }
+		if (! $error && ! $notrigger)
```

Comment:
For the "delete", the trigger must be called before the delete of object (can be after child tables) but must be before main table (this is the opposite for create and update).

Reply:
@eldy 
if the trigger is before, the member is always considered to be present in the member type !